ehealthid-cli/src/main/java/com/oviva/ehealthid/cli/MTlsRefreshCommand.java

```diff
@@ -0,0 +1,130 @@
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Callable;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "tls-refresh",
+    mixinStandardHelpOptions = true,
+    description = "Generate or Refresh the mTLS certificates in the Signing Key.")
+public class MTlsRefreshCommand implements Callable<Integer> {
+
+  @CommandLine.Option(
+      names = {"-i", "--iss", "--issuer-uri"},
+      description = "the issuer uri of the 'Fachdienst' identiy provider",
+      required = false)
+  private URI issuerUri;
+
+  @CommandLine.Option(
+      names = {"-e", "--existing"},
+      description = "the existing signing key",
+      required = true)
+  private String existingKey;
+
+  @CommandLine.Option(
+      names = {"-r", "--refresh"},
+      description = "refresh the existing mTLS certificate",
+      defaultValue = "false")
+  private boolean refresh;
+
+  private static final Logger logger = LoggerFactory.getLogger(MTlsRefreshCommand.class);
+
+  public Integer call() throws Exception {
+
+    var sigName = "sig";
+
+    logger.atInfo().log("using existing signing key '%s'".formatted(existingKey));
+    var inputStream = Files.newInputStream(Path.of(existingKey));
+    var jwks = JWKSet.load(inputStream);
+    var key = jwks.getKeys().get(0);
+    if (refresh) {
+      var newKey = generateCertificate(key);
+      System.out.println(newKey);
+      logger.atInfo().log("refreshing mTLS certificate");
+    } else {
+      var newKey = generateCertificateFromExistingKey(key);
+      logger.atInfo().log("generating mTLS certificate");
+      System.out.println(newKey);
+    }
+    return 0;
+  }
+
+  private JWK generateCertificateFromExistingKey(JWK key)
+      throws JOSEException, IOException, OperatorCreationException, CertificateEncodingException {
+    // The idea here is to give users that currently have keys, that do not have
+    // the relevant x5c field, the ability to generate a certificate for mTLS
+    // without going through the fuzz of generating a new key and talking to
+    // Gematik/BfArM about it. Which just generates work for everyone.
+
+    var now = Instant.now();
+    var nbf = now.minus(Duration.ofDays(1));
+    var exp = now.plus(Duration.ofDays(180));
+
+    var cert =
+        X509CertificateUtils.generateSelfSigned(
+            new Issuer(issuerUri),
+            Date.from(nbf),
+            Date.from(exp),
+            key.toECKey().toPublicKey(),
+            key.toECKey().toPrivateKey());
+
+    return new ECKey.Builder(key.toECKey())
+        .x509CertChain(List.of(Base64.encode(cert.getEncoded())))
+        .build();
+  }
+
+  private JWK generateCertificate(JWK key)
+      throws JOSEException, IOException, OperatorCreationException, CertificateEncodingException {
+    // TODO: This is one huge playground method
```

Comment:
That's the starting point I used: https://connect2id.com/products/nimbus-oauth-openid-connect-sdk/examples/utils/self-signed-client-cert